.changesets/config_simon_router_version_in_cache_key.md

```diff
@@ -0,0 +1,7 @@
+### Distributed query plan cache keys include the Router version number ([PR #6406](https://github.com/apollographql/router/pull/6406))
+
+If you have enabled [Distributed query plan caching](https://www.apollographql.com/docs/router/configuration/distributed-caching/#distributed-query-plan-caching), starting with this release and going forward you should anticipate additional cache regeneration cost when updating between any Router versions.
+
+More often than not, an Apollo Router release may contain changes that affect what query plans are generated or how they’re represented. To avoid using outdated entries from distributed cache, the cache key includes a counter that was manually incremented with relevant data structure or algorithm changes. Instead the cache key now includes the Router version number, so that different versions will always use separate cache entries.
```

Comment:
I think this doesn’t need to mention `CACHE_KEY_VERSION` v.s. `FEDERATION_VERSION` so I tried to pick a relevant level of detail.

Reply:
I would put this paragraph first. The point is that almost every release has had a cache key change, which comes with the additional cache regeneration cost. If people read the first paragraph, they'll think this is _more_ than what's currently happening, but it effectively isn't.